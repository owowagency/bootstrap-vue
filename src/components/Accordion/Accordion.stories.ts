import Accordion from '.';
import AccordionItem from '@/components/AccordionItem';

export default {
    title: 'Components/Accordion',
    component: Accordion,
};

const template = (args) => ({
    components: {Accordion, AccordionItem},
    setup: () => ({args}),
    template: `
        <Accordion v-bind="args">
            <template #default="{parentId}">
                <AccordionItem
                    header="Accordion Item #1"
                    :parent-id="args.alwaysOpen ? undefined : parentId"
                >
                    <strong>This is the first item's accordion body.</strong> It is shown by default, until the collapse plugin adds the appropriate classes that we use to style each element. These classes control the overall appearance, as well as the showing and hiding via CSS transitions. You can modify any of this with custom CSS or overriding our default variables. It's also worth noting that just about any HTML can go within the <code>.accordion-body</code>, though the transition does limit overflow.
                </AccordionItem>

                <AccordionItem
                    header="Accordion Item #2"
                    :parent-id="args.alwaysOpen ? undefined : parentId"
                >
                    <strong>This is the second item's accordion body.</strong> It is hidden by default, until the collapse plugin adds the appropriate classes that we use to style each element. These classes control the overall appearance, as well as the showing and hiding via CSS transitions. You can modify any of this with custom CSS or overriding our default variables. It's also worth noting that just about any HTML can go within the <code>.accordion-body</code>, though the transition does limit overflow.
                </AccordionItem>
            </template>
        </Accordion>
    `,
});

export const Default = template.bind({});

export const Flush = template.bind({});

Flush.args = {
    flush: true,
};

export const AlwaysOpen = template.bind({});

AlwaysOpen.args = {
    alwaysOpen: true,
};
