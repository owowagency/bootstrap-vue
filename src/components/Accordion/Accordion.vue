<template>
    <div
        :id="id"
        class="accordion"
        :class="classes"
    >
        <!--
        @slot Displays the content of the accordion
        @binding {string} parent-id The id of the parent accordion
        -->
        <slot :parent-id="id" />
    </div>
</template>

<script lang="ts">
import useFlush, {flushProps} from '@/composables/useFlush';
import {computed} from 'vue';
import {idProps} from '@/composables/useId';
import useClasses from '@/composables/useClasses';
</script>

<script lang="ts" setup>
const props = defineProps({
    ...flushProps,
    ...idProps,
});

const {classes} = useClasses(computed(() => [
    useFlush(props.flush, 'accordion-{0}').flushClass.value,
]));
</script>

<docs>
```vue
<Accordion>
    <AccordionItem header="Header">
        Content
    </AccordionItem>
</Accordion>
```

## Imported Props

| Prop name | Description                            | Type    | Values | Default |
| --------- | -------------------------------------- | ------- | ------ | ------- |
| flush     | Use the flush variant of the component | boolean | -      | false   |
| id        | The id used for the component          | string  | -      | `uuid`  |
</docs>
