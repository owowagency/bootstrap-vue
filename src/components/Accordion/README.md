# Accordion

```vue
<Accordion>
    <AccordionItem header="Header">
        Content
    </AccordionItem>
</Accordion>
```

## Imported Props

| Prop name | Description                            | Type    | Values | Default |
| --------- | -------------------------------------- | ------- | ------ | ------- |
| flush     | Use the flush variant of the component | boolean | -      | false   |
| id        | The id used for the component          | string  | -      | `uuid`  |

## Slots

| Name    | Description                           | Bindings                                                |
| ------- | ------------------------------------- | ------------------------------------------------------- |
| default | Displays the content of the accordion | **parent-id** `string` - The id of the parent accordion |
