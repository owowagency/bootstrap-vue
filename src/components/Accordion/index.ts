import Accordion from './Accordion.vue';

export default Accordion;
