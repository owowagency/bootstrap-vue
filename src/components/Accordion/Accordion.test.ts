import Accordion from '.';
import {describe} from 'vitest';

describe('template', () => {
    componentRenderTest(Accordion, {
        // Force an id to prevent snapshots from failing because of uuid.
        props: {id: 'dont-change-please'},
    });

    componentSlotRenderTest(Accordion);

    // TODO: Currently unable to test if the slot receives the `parent-id` prop.

    componentWrapperClassTest(Accordion, {flush: true}, 'accordion-flush');
});
