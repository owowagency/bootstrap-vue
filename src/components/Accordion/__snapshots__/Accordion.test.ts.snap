// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`template > renders default 1`] = `
"<div id=\\"dont-change-please\\" class=\\"accordion\\">
  <!--
        @slot Displays the content of the accordion
        @binding {string} parent-id The id of the parent accordion
        -->
</div>"
`;
